silq/pulses/pulse_types.py

```diff
@@ -667,6 +667,121 @@ def get_voltage(self, t: Union[float, Sequence]) -> Union[float, np.ndarray]:
+                                            unit='Hz', set_cmd=None,
+                                            vals=vals.Numbers())
+        self.sideband_mode = Parameter(initial_value=sideband_mode, set_cmd=None,
+                                       vals=vals.Enum('IQ', 'double'))
+        self.phase_reference = Parameter(initial_value=phase_reference,
+                                         set_cmd=None, vals=vals.Enum('relative',
+                                                                      'absolute'))
+        self._connect_parameters_to_config(
+            ['frequencies', 'phases', 'power', 'amplitudes', 'offset',
+             'frequency_sideband', 'sideband_mode', 'phase_reference'])
+
+        if self.sideband_mode is None:
+            self.sideband_mode = 'IQ'
+        if self.phase_reference is None:
+            self.phase_reference = 'relative'
+        if self.offset is None:
+            self.offset = 0
+
+    def __repr__(self):
+        properties_str = ''
+        try:
+            properties_str = f'power={self.power} dBm'
+            properties_str += f', A={self.amplitudes} V'
+            properties_str += f', f={self.frequencies} Hz'
+            properties_str += f', phases={self.phases} deg'
+            properties_str += '(rel)' if self.phase_reference == 'relative' else '(abs)'
+            if self.offset:
+                properties_str += f', offset={self.offset} V'
+            if self.frequency_sideband is not None:
+                properties_str += f'f_sb={freq_to_str(self.frequency_sideband)} ' \
+                                  f'{self.sideband_mode}'
+            properties_str += f', t_start={self.t_start}'
+            properties_str += f', duration={self.duration}'
+        except:
+            pass
+
+        return super()._get_repr(properties_str)
+
+    def get_voltage(self, t: Union[float, Sequence]) -> Union[float, np.ndarray]:
+        """Get voltage(s) at time(s) t.
+
+        Raises:
+            AssertionError: not all ``t`` between `Pulse`.t_start and
+                `Pulse`.t_stop
+        """
+        assert is_between(t, self.t_start, self.t_stop),\
+            f"voltage at {t} s is not in the time range " \
+            f"{self.t_start} s - {self.t_stop} s of pulse {self}"
+
+        if self.phase_reference == 'relative':
+            t = t - self.t_start
+
+        assert self.amplitudes is not None, f'Pulse {self.name} does not have ' \
+                                            f'specified amplitudes.'
+        assert self.frequencies is not None, f'Pulse {self.name} does not have ' \
+                                             f'specified frequencies.'
+        assert self.phases is not None, f'Pulse {self.name} does not have specified phases.'
+
+        assert len(self.amplitudes) == len(self.frequencies) == len(self.phases), \
+            f'Pulse {self.name} does not have equal number of amplitudes, frequencies and phases.'
+
+        waveform = 0.0
```

Comment:
create an empty array. Otherwise if amplitudes etc. is an empty list, it would return a float instead of an array

Reply:
done